### ISSUE TITLE
Schema Registry client and SSL

### ISSUE THREAD

**Issue by dodysw** [confluentinc/confluent-kafka-python#90]
Would it be possible to enhance the new AvroConsumer/AvroProducer/CachedSchemaRegistryClient to optionally accept certs related parameters so we can connect to Schema Registry on SSL channel that requires client certificate authentication. Or possibly accept a requests.Session() object (I think we can put a persistent certificate parameter to session object)

**Comment by ewencp** [comment]
@roopahc @criccomini Any thoughts on this? I'm a bit skeptical of exposing implementation details like the use of `requests` in the public interface. Is it possible to achieve something similar just by forwarding `*args` and `**kwargs` maybe?

**Comment by criccomini** [comment]
Agree on not wanting to expose `requests`.

**Comment by edenhill** [comment]
@dodysw Can you help us identify exactly what parameters are needed? Is certs= and verify= enough?

I would prefer to see this added as config dict properties rather than constructor args.

**Comment by dodysw** [comment]
For python requests, those are pretty much what needed. It looks like this:
```
ssl_cafile = '/opt/logs/sandbox/cert/kafka/ca-cert.pem'
ssl_certfile = '/opt/logs/sandbox/cert/kafka/client-cert.pem'
ssl_keyfile = '/opt/logs/sandbox/cert/kafka/client-key-plain.pem'
s = requests.Session()
s.cert = (ssl_certfile, ssl_keyfile)
s.verify = ssl_cafile
```
Client private key pem file must be unencrypted so it doesn't take a password.

**Comment by edenhill** [comment]
Okay, cool. What about the following new config properties:
 * `schema.registry.ssl.ca.location` = str
 * `schema.registry.ssl.certificate.location` = str | tuple
 * `schema.registry.ssl.key.location` = str

This is in line with existing SSL config properties for Kafka.
(Search for ssl.ca.location here: https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md)

**Comment by dodysw** [comment]
Close, I think `schema.registry.ssl.certificate.location` shouldn't be a tuple if `schema.registry.ssl.key.location` is there.

**Comment by roopahc** [comment]
@ewencp  Should be possible since we are hitting rest end points after all.   We can take them as argument to AvroProducer, AvroConsumer classes

**Comment by oh54** [comment]
Commenting to say that I would very much be interested in being able to turn off server certificate verification when using Schema Registry over HTTPS.

**Comment by rnpridgeon** [comment]
Would setting `REQUESTS_CA_BUNDLE` work here? 

http://docs.python-requests.org/en/master/user/advanced/

>This list of trusted CAs can also be specified through the REQUESTS_CA_BUNDLE environment variable.

**Comment by madtimber** [comment]
Am I correct in reading thru this, that there is no current way to use this library with a schema registry over HTTPS?

@rnpridgeon I notice you self assigned this issue to yourself. Are you currently working on adding this functionality?

**Comment by madtimber** [comment]
So I'm realizing now that my issue is different from @dodysw. I don't need to do client cert authentication, just need to verify the server's cert. And there's a way to handle that.

My solution, for anyone else searching for something similar, was to set the `REQUESTS_CA_BUNDLE` environment variable with our combined PEMs from our truststore.

It's a single line, easily missed, in the `requests` documentation:

```
This list of trusted CAs can also be specified through the REQUESTS_CA_BUNDLE environment variable.
```
Full docs here: http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification

**Comment by rnpridgeon** [comment]
fixed with #408
